From 533a45da1a8900267a667648450976bc334b71a8 Mon Sep 17 00:00:00 2001
From: Namjae Jeon <namjae.jeon@samsung.com>
Date: Wed, 26 May 2021 15:54:25 +0900
Subject: [PATCH] cifsd: add default case in switch statment in
 alloc_shash_desc()
Git-commit: 533a45da1a8900267a667648450976bc334b71a8
Patch-mainline: v5.15-rc1
References: bsn#12
Modified-by-SEL: No


Add default case in switch statment in alloc_shash_desc().

Reviewed-by: Dan Carpenter <dan.carpenter@oracle.com>
Signed-off-by: Namjae Jeon <namjae.jeon@samsung.com>
Signed-off-by: Steve French <stfrench@microsoft.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 fs/cifsd/crypto_ctx.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/fs/cifsd/crypto_ctx.c b/fs/cifsd/crypto_ctx.c
index 1830ae1b5ed3..dbfe36ee9be1 100644
--- a/fs/cifsd/crypto_ctx.c
+++ b/fs/cifsd/crypto_ctx.c
@@ -88,6 +88,8 @@ static struct shash_desc *alloc_shash_desc(int id)
 	case CRYPTO_SHASH_MD5:
 		tfm = crypto_alloc_shash("md5", 0, 0);
 		break;
+	default:
+		return NULL;
 	}
 
 	if (IS_ERR(tfm))
-- 
2.33.0

